coalib/bearlib/languages/Language.py

```diff
@@ -429,14 +429,40 @@ def limit_versions(language, limit, operator):
         versions = [version for version in language.versions
                     if operator(int(version), limit)]
     else:
-
         versions = [version for version in language.versions
                     if operator(version, limit)]
     if not versions:
         raise ValueError('No versions left')
     return type(language)(*versions)
 
 
+def compare(language, comparable, operator):
+    """
+    Compares a Language instance depending on the type of comparable and the
+    given operator:
+
+    >>> Language['C'] == Language["C"]
+    True
+    >>> Language['py 2'] == Language['py 3']
+    False
+
+    :param language:
+        A `Language` instance.
+    :param comparable:
+        An instance to be compared to `language`.
+    :param operator:
+        The operator to use for the comparison.
+    :return:
+        True/False for same type comparison or a new `Language` instance with
+        limited versions.
+    """
+    if isinstance(comparable, Language):
+        return type(language) is type(comparable) and operator(
+            language.versions, comparable.versions)
```

Comment:
This simple `.versions` comparison creates confusions, especially with `<` and `>`, since for example `(2.7, 3.6) < (3.3, 3.4)`

We need to find some new strategy here...

**CC** @sils